sqlitecluster/SQLiteNode.cpp

```diff
@@ -1779,14 +1926,44 @@ void SQLiteNode::_changeState(SQLiteNode::State newState) {
     // Exclusively lock the stateMutex, nobody else will be able to get a shared lock until this is released.
     unique_lock<decltype(stateMutex)> lock(stateMutex);
 
-    // We send any unsent transactions here before we finish switching states. Normally, this does nothing, unless
-    // we're switching down from LEADING or STANDINGDOWN, but we need to make sure these are all sent to the new
-    // leader before we complete the transition.
-    _sendOutstandingTransactions();
-
     // Did we actually change _state?
     State oldState = _state;
     if (newState != oldState) {
+        // If we were following, and now we're not, we give up an any replications.
+        if (oldState == FOLLOWING) {
+            {
+                SINFO("Stopping following locking");
+                unique_lock<mutex> lock(_replicationCommandMutex);
+                SINFO("Stopping following locked");
+                // Clear the replication queue.
+                while (true) {
+                    try {
+                        _replicationCommands.pop();
+                    } catch (const out_of_range& e) {
+                        break;
+                    }
+                }
+
+                // Kill the threads until we're following again.
+                _replicationThreadsShouldExit = true;
+            }
+            _replicationCV.notify_all();
+            for (auto& t : _replicationThreads) {
+                t.join();
+            }
+            _replicationThreads.clear();
+
+            // Done exiting.
+            _replicationThreadsShouldExit = false;
+        }
+        if (newState == FOLLOWING) {
+            // Start the replication threads.
```

Comment:
I'd suggest moving this comment up a line (and adding an empty line), so it's like:
```

// If are starting to follow, start all our replication threads
if (newState == FOLLOWING) {
   ...
```